src/components/hooks/useWS.ts

```diff
@@ -15,13 +16,29 @@ const useWS = <T extends TSocketEndpointNames>(name: T) => {
     const send = useCallback(
         async (data?: Parameters<typeof apiManager.augmentedSend<T>>[1]) => {
             setIsLoading(true)
-            try {
-                const response = await apiManager.augmentedSend(name, data)
-                setData(response[name] as TSocketResponseData<T>)
-            } catch (e) {
-                setError(e)
-            } finally {
-                setIsLoading(false)
+            const readyState = parseInt(sessionStorage.getItem('websocket_ready_state'))
+            if (readyState === 1 || readyState === 0) {
+                try {
+                    const response = await apiManager.augmentedSend(name, data)
+                    setData(response[name] as TSocketResponseData<T>)
+                } catch (e) {
+                    setError(e)
+                } finally {
+                    setIsLoading(false)
+                }
+            } else {
+                if (isBrowser()) {
+                    const currentLanguage = getLanguage() ?? 'en'
```

Comment:
Can't `getLanguage()` return a default lang? Why?
I think that instead of such checks you should be able to return a default.

Reply:
its old method which we are using to get language  so i didn't change the method

Reply:
You can improve it. Why not.

Reply:
thats where the high test coverage helps :) to increase confidence and thus enable refactors like that ;)
so developers are not worried of changing stuff as they know its gonna be caught by tests